db/migrate/20210621002201_add_columns_to_email_log_to_match_incoming_for_smtp_imap.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+class AddColumnsToEmailLogToMatchIncomingForSmtpImap < ActiveRecord::Migration[6.1]
+  def up
+    add_column :email_logs, :cc_addresses, :text, null: true
+    add_column :email_logs, :cc_user_ids, :integer, array: true, null: true
+    add_column :email_logs, :raw, :text, null: true
+    add_column :email_logs, :topic_id, :integer, null: true
+
+    add_index :email_logs, :topic_id
```

Comment:
Are we able to add this as a partial index on `topic_id IS NOT NULL`? From memory, the `email_logs` table is one of the largest table on some sites so we don't want to be carrying a big index around.

Reply:
Doesn't that just happen automatically if you are adding an index to a NULLable column? Why would it index the NULL values?

Reply:
null values are still indexed by default to support queries like `where topic_id IS NULL`. See https://github.com/discourse/discourse/blob/53dab8cf1ecf08979c0f4c17fdd8f2181b77b5ba/app/models/post.rb#L1178-L1179 for the partial indexes we use in Discourse even if a column is nullable